File: example implementations/java/convertor/src/rifflike/MagicaChunk.java
package rifflike;

import utils.EL;
import static utils.NumberUtilities.sum;

public class MagicaChunk {
    public String id;
    public int size;
    public int subchunkSize;

    public byte[] content;
    public EL<MagicaChunk> subChunks;


    public MagicaChunk(String id, byte[] bytes) {
        this(id, bytes, new EL<>());
    }
    public MagicaChunk(String id, byte[] bytes, EL<MagicaChunk> subChunks) {
        this.id = id;
        this.size = bytes.length;
        this.content = bytes;
        this.subChunks = subChunks;
        subchunkSize = sum(subChunks.convertAll(c -> c.size + 12));
    }

    public EL<MagicaChunk> getSubChunks(String id) {
        return subChunks.allMatches(c -> c.id.equals(id));
    }

    public MagicaChunk getSubChunk(String id) {
        return subChunks.firstMatch(c -> c.id.equals(id));
    }
}

File: example implementations/csharp/cvox-convertor/rifflike/MagicaChunk.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using cvox_convertor.utils;

namespace cvox_convertor.rifflike
{
    public class MagicaChunk
    {
        public string Id;
        public int Size;
        public int SubchunkSize;

        public byte[] Content;
        public List<MagicaChunk> SubChunks;

        public MagicaChunk(string id, byte[] bytes) : this(id, bytes, new()) { }

        public MagicaChunk(string id, byte[] bytes, List<MagicaChunk> subChunks)
        {
            Id = id;
            Size = bytes.Length;
            Content = bytes;
            SubChunks = subChunks;
            SubchunkSize = subChunks.ConvertAll(c => c.Size + 12).Sum();
        }

        public List<MagicaChunk> GetSubChunks(string id)
        {
            return SubChunks.AllMatches(c => c.Id == id);
        }

        public MagicaChunk getSubChunk(string id)
        {
            return SubChunks.FirstMatch(c => c.Id == id);
        }
    }
}
